src/method/method_callback.cpp

```diff
@@ -176,6 +177,13 @@ namespace cubmethod
 		m_qid_handler_map[qid] = request.handler_id;
 	      }
 	  }
+	else
```

Comment:
This is a safeguard in case the query execution fails for any reason. Even if there is no code to clear all statement handlers for SP queries (`cubmethod::get_callback_handler ()->free_query_handle_all (true);`), this safeguard fixes the problem.

Reply:
OK, Thank you.